src/jemalloc.c

```diff
@@ -930,92 +937,124 @@ malloc_slow_flag_init(void) {
+				const char *linkname =
 #  ifdef JEMALLOC_PREFIX
-			    "/etc/"JEMALLOC_PREFIX"malloc.conf"
+				    "/etc/"JEMALLOC_PREFIX"malloc.conf"
 #  else
-			    "/etc/malloc.conf"
+				    "/etc/malloc.conf"
 #  endif
-			    ;
+				    ;
 
-			/*
-			 * Try to use the contents of the "/etc/malloc.conf"
-			 * symbolic link's name.
-			 */
+				/*
+				 * Try to use the contents of the
+				 * "/etc/malloc.conf" symbolic link's name.
+				 */
 #ifndef JEMALLOC_READLINKAT
-			linklen = readlink(linkname, buf, sizeof(buf) - 1);
+				linklen = readlink(linkname, buf, PATH_MAX);
 #else
-			linklen = readlinkat(AT_FDCWD, linkname, buf,
-			    sizeof(buf) - 1);
+				linklen = readlinkat(AT_FDCWD, linkname, buf,
+				    PATH_MAX);
 #endif
-			if (linklen == -1) {
-				/* No configuration specified. */
-				linklen = 0;
-				/* Restore errno. */
-				set_errno(saved_errno);
-			}
+				if (linklen == -1) {
+					/* No configuration specified. */
+					linklen = 0;
+					/* Restore errno. */
+					set_errno(saved_errno);
+				}
 #endif
-			buf[linklen] = '\0';
-			opts = buf;
-			break;
-		} case 3: {
-			const char *envname =
+				buf[linklen] = '\0';
+				opts_cache[i] = buf;
+				break;
+			} case 3: {
+				const char *envname =
 #ifdef JEMALLOC_PREFIX
-			    JEMALLOC_CPREFIX"MALLOC_CONF"
+				    JEMALLOC_CPREFIX"MALLOC_CONF"
 #else
-			    "MALLOC_CONF"
+				    "MALLOC_CONF"
 #endif
-			    ;
-
-			if ((opts = jemalloc_secure_getenv(envname)) != NULL) {
-				/*
-				 * Do nothing; opts is already initialized to
-				 * the value of the MALLOC_CONF environment
-				 * variable.
-				 */
-			} else {
-				/* No configuration specified. */
-				buf[0] = '\0';
-				opts = buf;
+				    ;
+
+				if ((opts_cache[i] =
+				    jemalloc_secure_getenv(envname)) != NULL) {
+					/*
+					 * Do nothing; opts is already
+					 * initialized to the value of the
+					 * MALLOC_CONF environment variable.
+					 */
+				} else {
+					/* No configuration specified. */
+					opts_cache[i] = empty_opts;
+				}
+				break;
+			} default:
+				not_reached();
+				opts_cache[i] = empty_opts;
 			}
-			break;
-		} default:
-			not_reached();
-			buf[0] = '\0';
-			opts = buf;
+			assert(opts_cache[i] != NULL);
+		}
+		opts = opts_cache[i];
+		if (!search_confirm_conf_only && opt_confirm_conf) {
+			malloc_printf("<jemalloc>: malloc_conf (%s): \"%s\"\n",
+			    opts_explain[i], opts);
 		}
 
 		while (*opts != '\0' && !malloc_conf_next(&opts, &k, &klen, &v,
 		    &vlen)) {
+
+#define CONF_ERROR(msg, k, klen, v, vlen)				\
+			if (!search_confirm_conf_only) {		\
+				malloc_conf_error(			\
+				    msg, k, klen, v, vlen);		\
+				error = true;				\
+			}
+#define CONF_CONTINUE							\
+			if (!search_confirm_conf_only			\
+			    && opt_confirm_conf && !error) {		\
+				malloc_printf(				\
+				    "<jemalloc>: Set conf value: "	\
+				    "%.*s:%.*s\n", (int)klen, k,	\
+				    (int)vlen, v);			\
+				continue;				\
+			} else {					\
+				continue;				\
```

Comment:
Let's only have 1 `continue` statement (move it outside of the branch). Also `{}` the whole thing.